aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2052,26 +2052,30 @@ Tensor backward_analytic_function_of_a_matrix(
 // Computing the Matrix Exponential with an Optimized Taylor Polynomial Approximation.
 // Mathematics 2019, 7, 1174.
 //
-Tensor matrix_exp(const Tensor& a) {
-  TORCH_CHECK(a.dim() >= 2
-          && (at::isFloatingType(a.scalar_type())
-           || at::isComplexType(a.scalar_type())),
-              "matrix_exp(", a.scalar_type(), "{", a.sizes(), "}): expected a tensor "
-              "of floating or complex types with dim at least 2");
-  TORCH_CHECK(a.size(-1) == a.size(-2),
-              "matrix_exp(", a.scalar_type(), "{", a.sizes(), "}): expected a tensor "
-              "of squared matrices");
+Tensor linalg_matrix_exp(const Tensor& a) {
+  squareCheckInputs(a);
+  TORCH_CHECK((at::isFloatingType(a.scalar_type()) || at::isComplexType(a.scalar_type())),
+              "Expected a floating point or complex tensor as input. Got: ", a.scalar_type());
```

Comment:
What if we introduce additional parameter to squareCheckInputs indicating the name of the function it fails in if that happens? Same goes for other checks. Will also make the errors coming from `torch.linalg` a touch easier to parse and identify.

Reply:
I think that's a great idea. I'll do that in a bit.

Reply:
Done! Could you review it please? https://github.com/pytorch/pytorch/pull/62734

Reply:
The dtype error message should also start with the name of the operator, like "torch.linalg.matrix_exp: Expected..."

Reply:
This is done in the third PR of this stack.